src/v/kafka/server/kafka_probe.h

```diff
@@ -74,6 +75,34 @@ class kafka_probe {
           },
           {},
           {sm::shard_label});
+
+        for (auto compress_type : model::all_batch_compression_types) {
+            auto compress_type_name = fmt::format("{}", compress_type);
+            auto compression_label = sm::label("compression_type")(
+              compress_type_name);
+
+            _metrics.add_group(
+              "kafka",
+              {
+                sm::make_counter(
+                  "produced_bytes",
```

Comment:
Wondering whether this name will make people use this metric for other stuff than "compression type analysis".

Reply:
I guess that's "fine"? It would work as total bytes produced if you just aggregate away the label ... though perhaps it's redundant in that sense with one you added earlier?

Happy to change it. The struggle is that it's sort of a guideline to have the metric name be _what_ it is measuring, in this case produced bytes and not refer divisions to the labels by the labels (like "by compression", "by handler", "by shard", etc) since labels may be aggregated away at any point e.g., in Redpanda or in your prom query, making the explicit mention of the label redundant.

Really though I'm happy to name this whatever: what are you thoughts?

Reply:
> The struggle is that it's sort of a guideline to have the metric name be what it is measuring

Yeah I guess all my suggestions would go against this. I am not sure I fully agree with it but I don't have a strong opinion on this in general so happy to leave as is.